addons/website_sale_stock_wishlist/views/templates.xml

```diff
@@ -5,10 +5,45 @@
             <small class="text-danger d-md-block" t-if="wish.product_id.sudo()._is_sold_out()">Temporarily out of stock</small>
         </xpath>
         <xpath expr="//button[hasclass('o_wish_rm')]" position="after">
-            <t t-set="notify" t-value="wish.stock_notification"/>
-            <button groups="base.group_user,base.group_portal" t-att-data-notify="notify" t-if="notify or wish.product_id.sudo()._is_sold_out()" type="button" class="btn btn-link o_notify_stock no-decoration">
-                <small><i t-attf-class="fa #{'fa-check-square-o' if notify else 'fa-square-o'}"></i> Be notified when back in stock</small>
-            </button>
+            <t t-if="not wish.product_id.allow_out_of_stock_order and wish.product_id._is_sold_out()">
+                <div id="stock_notification_div" t-if="not wish.product_id.allow_out_of_stock_order"
+                     class="small d-inline-block">
+                    <div class="btn btn-link" t-if="not wish.product_id._has_stock_notification(wish.partner_id)"
+                         id="wishlist_stock_notification_message">
+                        <small>
+                            <i class="fa fa-envelope-o"/>
+                            Get notified when back in stock
+                        </small>
+                    </div>
+                    <div id="stock_notification_form" class="d-none">
+                        <div class="input-group">
+                            <input id="stock_notification_input"
+                                    class="input-group-append"
+                                   name="email"
+                                   type="text"
+                                   placeholder="youremail@gmail.com"
+                                   t-att-value="request.env.user.partner_id.email"/>
```

Comment:
Should this not use the session "email" if available? AFAIK this is also valid for public users?